File: mcpservers/llmwork_wechatpublic/pkg/mcp/protocol.go
package mcp

// MCPRequest MCP 请求结构
type MCPRequest struct {
	JSONRPC string                 `json:"jsonrpc"` // 必须是 "2.0"
	ID      interface{}            `json:"id"`      // 请求 ID
	Method  string                 `json:"method"`
	Params  map[string]interface{} `json:"params,omitempty"`
}

// MCPResponse MCP 响应结构
type MCPResponse struct {
	JSONRPC string      `json:"jsonrpc"` // 必须是 "2.0"
	ID      interface{} `json:"id"`      // 对应请求的 ID
	Result  interface{} `json:"result,omitempty"`
	Error   *MCPError   `json:"error,omitempty"`
}

// MCPError MCP 错误结构
type MCPError struct {
	Code    int    `json:"code"`
	Message string `json:"message"`
}

// Tool MCP 工具定义
type Tool struct {
	Name        string      `json:"name"`
	Description string      `json:"description"`
	InputSchema InputSchema `json:"inputSchema"`
}

// InputSchema 工具输入模式
type InputSchema struct {
	Type       string              `json:"type"`
	Properties map[string]Property `json:"properties,omitempty"`
	Required   []string            `json:"required,omitempty"`
}

// Property 属性定义
type Property struct {
	Type        string   `json:"type"`
	Description string   `json:"description,omitempty"`
	Enum        []string `json:"enum,omitempty"`
	Items       *Items   `json:"items,omitempty"`
}

// Items 数组项定义
type Items struct {
	Type string `json:"type"`
}

// ToolResult 工具调用结果
type ToolResult struct {
	Content []Content `json:"content"`
	IsError bool      `json:"isError,omitempty"`
}

// Content 内容定义
type Content struct {
	Type string `json:"type"`
	Text string `json:"text"`
}

File: lib/features/mcp/builtins/servers/protocol.dart
/// MCP 协议类型定义
class McpRequest {
  final dynamic id;
  final String method;
  final Map<String, dynamic>? params;

  McpRequest({this.id, required this.method, this.params});

  factory McpRequest.fromJson(Map<String, dynamic> json) {
    return McpRequest(
      id: json['id'],
      method: json['method'] as String,
      params: json['params'] as Map<String, dynamic>?,
    );
  }
}

class McpResponse {
  final dynamic id;
  final dynamic result;
  final Map<String, dynamic>? error;

  McpResponse({this.id, this.result, this.error});

  Map<String, dynamic> toJson() {
    final json = <String, dynamic>{
      'jsonrpc': '2.0',
      'id': id,
    };
    if (result != null) json['result'] = result;
    if (error != null) json['error'] = error;
    return json;
  }
}
